packages/nimble-components/src/accordion/specs/README.md

```diff
@@ -0,0 +1,210 @@
+        <div slot="heading">Accordion three</div>
+        <nimble-checkbox></nimble-checkbox>
+        <nimble-accordion>
+            <nimble-accordion-item>
+                <div slot="heading">Nested Accordion One</div>
+                Nested Accordion One Content
+            </nimble-accordion-item>
+            <nimble-accordion-item>
+                <div slot="heading">Nested Accordion Two</div>
+                Nested Accordion Two Content
+            </nimble-accordion-item>
+        </nimble-accordion>
+    </nimble-accordion-item>
+    <nimble-accordion-item>
+        <div slot="heading">Accordion four</div>
+        <nimble-button>Text Button</nimble-button>
+        Accordion four content
+    </nimble-accordion-item>
+</nimble-accordion>
+```
+
+### API
+
+#### Nimble Accordion
+
+[FAST accordion API](https://github.com/microsoft/fast/blob/57f3c22c6341d8a21d48b1ffb7fcbfab1ffd02d8/packages/web-components/fast-foundation/src/accordion/accordion.spec.md)
+
+-   _Component Name:_ `nimble-accordion`
+-   _Properties/Attributes:_
+    -   `appearance` - `block`, `outline`, `ghost`
+    -   FAST's attribute `expand-mode` will be omitted from the FAST base class due to issues discussed below.
+-   _Methods:_ Unchanged
+-   _Events:_ Unchanged
+-   _CSS Classes and Custom Properties that affect the component:_ Unchanged
+-   _Slots:_ Unchanged
+
+The design includes appearances of the accordion header that reflect those of the `block`, `outline`, and `ghost` appearances used in other components. These designs are similar to those of the nimble-button and its `appearance` types, but the accordion does not use the `.control` css class, it uses `.region` and `.heading`. Because of this, differences in border functionality, and the possibility of excess overridden css classes if patterns are used, shared styles will not be used, and new styles for the accordion and accordion-item will be created. The appearances of `block`, `outline`, and `ghost` will be implemented under the HTML attribute `appearance`, which will use conditional css styles based on `appearance`'s value (ex. `:host([appearance='outline'])` will have styling for the outline button).
+
+Documentation will be added to advise against using multiple Nimble Accordions with different appearances next to each other.
+
+When using nested accordions, the parent accordion will have the selected `appearance` type, and all children accordions will be forced to have the `ghost` appearance type.
+
+`expand-mode` is a FAST attribute, but is being temporarily restricted / removed from then nimble-accordion due to issues with the `single` expand-mode. See "Future Work" for more info.
+
+#### Nimble Accordion Item
+
+[FAST accordion-item API](https://github.com/microsoft/fast/tree/57f3c22c6341d8a21d48b1ffb7fcbfab1ffd02d8/packages/web-components/fast-foundation/src/accordion-item)
+
+-   _Component Name:_ `nimble-accordion-item`
+-   _Properties/Attributes:_
+    -   `error-visible` - boolean
```

Comment:
After reading the [Accordion IxD](https://github.com/ni/nimble/blob/28e170974e54cb8a2f97d0804119c191f9f60369/packages/nimble-components/src/accordion/specs/IxD.md) (yet to be approved), one interesting description was that an `accordion-item` should represent an error state if one of its children was in an error state. This is interesting in that it could cause us to re-evaluate this API, if, say, we wanted this behavior to occur with no user configuration.

Often, accordion-items are meant to be layout components, and not have any technical association with their content. With the current proposed API along with the guidance given in the IxD, to achieve the desired behavior a client would have to (in code-behind):
- explicitly detect when the error state of a component with an `accordion-item` changed
- find the `accordion-item` parent(s) it resides in (could be more than one?)
- explicitly change the `error-visible` state in each parent

Instead, is it better to have the `accordion-item` do automatic detection of these DOM changes, and then update its visual state as needed, thus requiring no user configuration? There are some questions that we would need to answer before being comfortable with this:
- How would we handle non-Nimble components that offered an error state, which we would also want to be reflected by the `accordion-item`?
- Would we be comfortable with using a `MutationObserver` to manage the automatic notification?
- Is it even okay to take this control away from the user (forces the `accordion-item` to always have this visual state when a child Nimble component is an error state)? Does it make sense to offer an API to turn this auto-detection on or off?

Reply:
I think consistency with our other component behaviors would be for the applications doing the validation and specifying of error states.

Should be a common thing frameworks deal with (propagating error states up a tree like FormGroups in Angular, etc).

Similar to how we aren't providing hooks in the HTML 5 validation apis directly. You have to listen for those events and set error visible on the component yourself. Or in more apps today you listen for Angular's validation events. Also similar to how disabling a text field doesn't automatically disable slotted action buttons in the text field, that is application responsibility.

Certainly doesn't represent all state. I.e. selection state across tree-items in managed by the element. But I think we are pretty consistent that validation state is managed by the app and not the components. Trying to think of contrary examples. Also could be convinced this is an exception.